signingscript/src/signingscript/task.py

```diff
@@ -209,11 +210,25 @@ def build_filelist_dict(context):
     filelist_dict = {}
     messages = []
     for artifact_dict in context.task["payload"]["upstreamArtifacts"]:
+        authenticode_comment = artifact_dict.get("authenticode_comment")
+        if authenticode_comment and not any("authenticode" in fmt for fmt in artifact_dict["formats"]):
+            raise TaskVerificationError("Cannot use authenticode_comment without an authenticode format")
+
+        if authenticode_comment and not any(path.endswith(".msi") for path in artifact_dict["paths"]):
+            # Don't have to think about .zip and such unpacking for the comment
+            raise TaskVerificationError("There is no support for authenticode_comment outside of msi's at this time")
         for path in artifact_dict["paths"]:
             full_path = os.path.join(context.config["work_dir"], "cot", artifact_dict["taskId"], path)
             if not os.path.exists(full_path):
                 messages.append("{} doesn't exist!".format(full_path))
             filelist_dict[path] = {"full_path": full_path, "formats": _sort_formats(artifact_dict["formats"])}
+            if authenticode_comment and "msi" in path:
```

Comment:
I'm wondering if we can simply accept `authenticode_comment`, and pass it down, since `sign_authenticode_file` already checks to make sure the file is an msi.

Could we simply the changes to this function to just pass down the `authenticode_comment`, and not have to worry about `msi` or `path` ? Would we ever need to support more than 1 msi file in an `upstreamArtifact`, where we want to supply different `authenticode_comment`s per msi? If so, would it be bad to require another `upstreamArtifact` dict? In that case, `authenticode_comment` could always be a string.

Reply:
Yea, this was a premature complexity, I think.  Nothing needs it.